In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from google.colab import files
uploaded = files.upload()

df = pd.read_csv("train.csv")
df.head(8)

Saving train.csv to train (1).csv


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S

In [ ]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [ ]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

A quantidade de nulos em 'Age' não é irrisória, mas também não afeta muita coisa, 'Cabin' é praticamente toda nula, tendo em vista que 687/891 são nulos e a quantidade de nulos em 'Embarked' é irrisória.

In [ ]:
# Os nulos de 'Age' serão substituídos pela media adquirida com a função .describe
df=df.fillna({"Age":df["Age"].mean()})

# 'Cabin' será removida, pois tem pouquíssimas informações
df.drop('Cabin', axis=1, inplace=True)

# Os nulos de 'Embarked' serão substituídos pelo valor que mais aparece
df=df.fillna({"Embarked":df["Embarked"].mode()[0]})

In [ ]:
df['Embarked'].value_counts()

S    646
C    168
Q     77
Name: Embarked, dtype: int64

A maioria dos passageiros era de Southampton, com 644, sendo seguido por 168 de Cherbourg e 77 de Queenstown

In [ ]:
df['Sex'].value_counts()

male      577
female    314
Name: Sex, dtype: int64

Em sua maioria os passageiros eram homens, 577, e a minoria era das mulheres, com 314

In [ ]:
df=df.replace('male', 0)
df=df.replace('female', 1)
df=df.replace('S', 0)
df=df.replace('Q', 1)
df=df.replace('C', 2)

Com o gênero sendo representado por 1 para mulheres e 0 para homens e as cidades por 0, 1 e 2 os dados poderão ser levados em conta pela máquina.

In [ ]:
df.drop('PassengerId', axis=1, inplace=True)
df.drop('Ticket', axis=1, inplace=True)
df.drop('Name', axis=1, inplace=True)
df

Survived  Pclass  Sex        Age  SibSp  Parch     Fare  Embarked
0           0       3    0  22.000000      1      0   7.2500         0
1           1       1    1  38.000000      1      0  71.2833         2
2           1       3    1  26.000000      0      0   7.9250         0
3           1       1    1  35.000000      1      0  53.1000         0
4           0       3    0  35.000000      0      0   8.0500         0
..        ...     ...  ...        ...    ...    ...      ...       ...
886         0       2    0  27.000000      0      0  13.0000         0
887         1       1    1  19.000000      0      0  30.0000         0
888         0       3    1  29.699118      1      2  23.4500         0
889         1       1    0  26.000000      0      0  30.0000         2
890         0       3    0  32.000000      0      0   7.7500         1

[891 rows x 8 columns]

As colunas 'PassengerId', 'Ticket' e 'Name' serão retiradas, pois não influenciam na sobrevivência dos passageiros

In [ ]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split

In [ ]:
X=df.drop('Survived', axis=1)
y=df['Survived']

X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8)

clf=RandomForestClassifier()
clf.fit(X_train, y_train)

clf.score(X_test, y_test)

0.8491620111731844